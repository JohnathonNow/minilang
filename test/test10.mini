var PointT := object(:X, :Y)

var A := PointT(10, 20)

print('A:X = {A:X}\n')
print('A = {A}\n')

A:Y := 200

method(:string, PointT; P) 'point({P:X}, {P:Y})'

print('A = {A}\n')

method(::+, PointT, PointT; A, B) PointT(A:X + B:X, A:Y + B:Y)

var B := PointT(23, 9.1)

print('{A} + {B} = {A + B}\n')

method(::*, PointT, NumberT; A, N) PointT(A:X * N, A:Y * N)
method(::*, NumberT, PointT; N, A) PointT(N * A:X, N * A:Y)

print('{A} * 5 = {A * 5}\n')
print('7.1 * {B} = {7.1 * B}\n')