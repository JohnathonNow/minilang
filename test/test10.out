A:X = 10
A = point(X: 10, Y: 20)
A = point(10, 200)
point(10, 200) + point(23, 9.100000) = point(33, 209.100000)
point(10, 200) * 5 = point(50, 1000)
7.1 * point(23, 9.100000) = point(163.300000, 64.610000)
C = point(1, 2)
C = vector(1, 2, 3)
